mmaction/datasets/pipelines/augmentations.py

```diff
@@ -533,6 +533,85 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class RandomRescale(object):
+    """Randomly resize images so that the short_edge is resized to a specific
+    size in a given range. The scale ratio is unchanged after resizing.
+
+    Required keys are "imgs", "img_shape", "modality", added or modified
+    keys are "imgs", "img_shape", "keep_ratio", "scale_factor", "lazy",
+    "resize_size", "short_edge". Required keys in "lazy" is None, added or
+    modified key is "interpolation".
+
+    Args:
+        scale_range (Tuple[int]): The range of short edge length. A closed
+            interval.
+        interpolation (str): Algorithm used for interpolation:
+            "nearest" | "bilinear". Default: "bilinear".
+        lazy (bool): Determine whether to apply lazy operation. Default: False.
+    """
+
+    def __init__(self, scale_range, interpolation='bilinear', lazy=False):
+        self.scale_range = scale_range
+        # make sure scale_range is legal
+        assert len(scale_range) == 2
+        assert type(scale_range[0]) is int
+        assert type(scale_range[1]) is int
+        assert scale_range[0] < scale_range[1]
+
+        self.keep_ratio = True
+        self.interpolation = interpolation
+        self.lazy = lazy
+
+    def __call__(self, results):
```

Comment:
I'm not sure. If constructing a new `Resize` instance is not time-consuming, it will be OK.

Reply:
yeah there are some overhead, can this object be moved to init?

Reply:
No, since the target size is dynamically determined.

Reply:
How about initializing the `Resize` in `def __init__`, but changing the `resize.scale` in `__call__`.

Reply:
I think ur first version would be nice. Although it may introduce additional overhead, it simplifies the code and doesn't require explicitly setting attributes of an existing object. BTW, the overhead is very little comparing to the data loading pipeline.

Reply:
okay makes sense